app/src/main/java/com/ucsd/cse110t15m260/roommateinventory/AddItemActivity.java

```diff
@@ -0,0 +1,322 @@
+
+        mNameView = (EditText) findViewById(R.id.text_item_name);
+        mCategoryView = (EditText) findViewById(R.id.text_item_category);
+        mQuantityView = (EditText) findViewById(R.id.text_item_count);
+        mDescriptionView = (EditText) findViewById(R.id.text_description);
+
+        mUserName = (TextView) findViewById(R.id.text_username_created_by);
+
+        /* TODO: Make sure it's passed on the other end as InventoryItem  */
+        Serializable item =  getIntent().getSerializableExtra("InventoryItem");
+        TextView createdByTextView = (TextView) findViewById(R.id.text_created_by);
+        FloatingActionButton fab = (FloatingActionButton) findViewById(R.id.fab);
+
+        TextView usernameTextView = (TextView) findViewById(R.id.text_username_created_by);
+        if(item == null) {
+            /* Creation Mode: New item needs to be created */
+            theItem = new InventoryItem();
+
+
+            fab.setOnClickListener(new OnClickListener() {
+                @Override
+                public void onClick(View v) {
+                    attemptToCreateNewItem();
+                }
+            });
+
+            createdByTextView.setVisibility(GONE);
+            usernameTextView.setVisibility(GONE);
+            fab.setVisibility(VISIBLE);
+
+        } else {
+            /* Viewing Mode: item already exists. */
+            theItem = (InventoryItem) item;
+            fab.setVisibility(GONE);
+            createdByTextView.setVisibility(VISIBLE);
+            usernameTextView.setVisibility(VISIBLE);
+
+            if(theItem.getCreator() != null) {
+                usernameTextView.setText(theItem.getCreator().getUsername());
+            }
+
+        }
+
+
+        // ATTENTION: This was auto-generated to implement the App Indexing API.
+        // See https://g.co/AppIndexing/AndroidStudio for more information.
+        client = new GoogleApiClient.Builder(this).addApi(AppIndex.API).build();
+
+        //Register the listener for the captuer image button
+        ImageButton imageButton = (ImageButton) findViewById(R.id.image_button);
+        imageButton.setOnClickListener(new OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                launchCamera();
+            }
+        });
+
+
+    }
+
+
+    public void attemptToCreateNewItem() {
+        mNameView.setError(null);
+        mCategoryView.setError(null);
+        mQuantityView.setError(null);
+        mDescriptionView.setError(null);
+
+
+        boolean cancel = false;
+        View focusView = null;
+
+        String itemName = mNameView.getText().toString();
+        String category = mCategoryView.getText().toString();
+        Number quantity = Integer.parseInt(mQuantityView.getText().toString());
+        String description = mDescriptionView.getText().toString();
+
+        Log.d("AddItemActivity", "Before checking fields");
+        if (TextUtils.isEmpty(itemName)) {
```

Comment:
This screen is used in two cases, when adding a new item and when viewing an existing one. This logic determines which case is happening and acts accordingly. Because of that, @Jlsmith10 and I thought it belonged in the Activity.